app/assets/javascripts/code/services/google/_map.js

```diff
@@ -203,3 +222,76 @@ export const map = (
 
   return new Map();
 };
+
+
+function definePopupClass() {
+  /**
+   * A customized popup on the map.
+   * @param {!google.maps.LatLng} position
+   * @param {!Element} content
+   * @constructor
+   * @extends {google.maps.OverlayView}
+   */
+  Popup = function(position, content, colorClass, $rootScope, clicableObject) {
+    this.position = position;
+
+    this.anchor = document.createElement('div');
+    this.anchor.classList.add('fixed-marker');
+    this.anchor.classList.add(colorClass);
+    this.anchor.innerText = content;
+    this.anchor.addEventListener('click', function() {
+      console.log('CLICKED')
+      $rootScope.$broadcast('markerSelected', {session_id: clicableObject});
+    });
```

Comment:
Is there a way to add this event listener in a different way, different place? cause it requires $rootScope, and also it's coupled with session_id and we wanted this popup to be as general as possible.